pkg/armhelpers/azurestack/loganalytics.go

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+package azurestack
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	oi "github.com/Azure/azure-sdk-for-go/services/preview/operationalinsights/mgmt/2015-11-01-preview/operationalinsights"
+	om "github.com/Azure/azure-sdk-for-go/services/preview/operationsmanagement/mgmt/2015-11-01-preview/operationsmanagement"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+// EnsureDefaultLogAnalyticsWorkspace ensures the default log analytics exists corresponding to specified location
+func (az *AzureClient) EnsureDefaultLogAnalyticsWorkspace(ctx context.Context, resourceGroup, location string) (workspaceResourceID string, err error) {
```

Comment:
can this new file be unit tested as well? see disk.go and disk_test.go for an example of testing armhelpers functions. You should be able to use a mock client for this.

Reply:
@CecileRobertMichon, I look at the disk_test.go and I see these tests only for the azure stack. are you asking to do the same?

Reply:
> can this new file be unit tested as well? see disk.go and disk_test.go for an example of testing armhelpers functions. You should be able to use a mock client for this.

added the unit tests to cover these methods.